src/js/client/editprefset-genericSettingsTable-widget.js

```diff
@@ -0,0 +1,172 @@
+/* global lunr */
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.settingsTable");
+
+fluid.defaults("gpii.devpmt.genericSettingsTableWidget", {
```

Comment:
Explanation of the purpose of this component, layout of its options, invokers, and where its model relay is set up, as well as use made of the global "lunr" dependency

Reply:
This is documented now.